BaragonAgentService/src/main/java/com/hubspot/baragon/agent/managers/AgentRequestManager.java

```diff
@@ -225,6 +241,51 @@ public Response getRenderedConfigs(String serviceId) {
     return Response.ok(result).build();
   }
 
+
+  public static String getServiceBasePathWithoutLeadingSlash(String path){
+    if (path.startsWith("/")){
+      path = path.replaceFirst("/", "");
+    }
+    return path;
+  }
+
+  public Response purgeCache(String serviceId) {
+    LOG.info("purgeCache() called with serviceId={}", serviceId);
+
+    Optional<BaragonService> maybeService;
+    if (internalStateCache.containsKey(serviceId)){
+      maybeService = Optional.of(internalStateCache.get(serviceId).getService());
+    }
+    else {
+      maybeService = stateDatastore.getService(serviceId);
+    }
+
+    // 1. if no service exists, return an error response early
+    if (!maybeService.isPresent()){
+      return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(String.format("Could not find service with serviceId=%s", serviceId)).build();
+    }
+
+    // 2. remove a leading slash if it exists and generate the URI
+    String serviceBasePathWithoutLeadingSlash = getServiceBasePathWithoutLeadingSlash(
+        maybeService.get().getServiceBasePath()
+    );
+    String purgeCacheUri = String.format(
+        baragonAgentConfiguration.getPurgeCacheUriFormat(),
+        serviceBasePathWithoutLeadingSlash
+    );
+    LOG.info("purgeCache() uri={}", purgeCacheUri);
+
+    // 3. now build the request
+    final HttpRequest.Builder builder = HttpRequest.newBuilder()
+        .setUrl(purgeCacheUri)
+        .setMethod(Method.GET);
```

Comment:
Should we make a cache purge a POST rather than a GET?

Reply:
agreed, just need to check the purge module will accept POST requests